grid/chat/members/maps/discord.suma

```diff
@@ -0,0 +1,187 @@
+profile = "chat/members@1.0"
+provider = "discord"
+
+"""
+GetMembers map
+"""
+map GetMembers {
+  return map error if (input.workspace === undefined) {
+    title = "Input parameter `workspace` is required for provider discord"
+  }
+
+  http GET "/guilds/{input.workspace}/members" {
+    request "application/json" {
+      query {
+        after = input.page
+        limit = input.limit
+      }
+
+      headers {
+        "Authorization" = `Bot ${parameters.accessToken}`
+      }
+    }
+
+    response 200 "application/json" {
+      members = call MapMembers(members = body)
+      rateLimit = call MapRateLimitDetails(headers = headers)
+
+      result.members = members,
+      result.nextPage = members[members.length - 1].id
+
+      set if (rateLimit) {
+        result.rateLimit = rateLimit
+      }
+
+      map result result
+    }
+
+    response {
+      error = call MapDiscordError(
+        statusCode = statusCode,
+        error = body,
+        headers = headers,
+      )
+      
+      map error error
+    }
+  }
+}
+
+"""
+Operation MapMembers
+Returns array mapping members in received body
+"""
+operation MapMembers {
+  members = args.members.map(member => {
+    const resultMember = {
+      id: member.user.id,
+      joinedAt: Date.parse(member.joined_at),
+      isBot: member.user.bot ? true : false,
+    }
+
+    // map user's platform wide nickname - username
+    if (member.user && member.user.username) {
```

Comment:
`member.user` condition doesn't make sense if you assume it exists on previous lines. I'd just move this into the construction of `resultMember` object.

Reply:
Fair enough